src/io.c

```diff
@@ -379,6 +383,47 @@ static void __fs_flush(void)
     }
 }
 
+#define SYNC_FILE_RANGE
+
+#ifdef SYNC_FILE_RANGE
+#define CHUNK_SIZE	(512 * 1024 * 1024)
+/* NOTE: Using sync_file_range() function does not portable */
+void fs_sync(void)
+{
+    off_t offset = 0;
+    size_t chunk_size = CHUNK_SIZE;
+    size_t bytes_to_sync = 0;
+
+    /* request writeback */
+	while (offset < dev_size) {
+		bytes_to_sync = (dev_size - offset) < chunk_size ?
+			(dev_size - offset) : chunk_size;
+
+		// Optionally, you can also use sync_file_range to wait for the write to complete
+		if (sync_file_range(fd, offset, dev_size,
```

Comment:
if the length of sync_file_range is dev_size, it request a write and wait for a whole partition.

Reply:
fixed